src/client/components/network-import/excel-import-wizard.js

```diff
@@ -210,20 +211,40 @@ class ExcelImportSubWizard extends React.Component {
 
   async handleFinish() {
     const { networkName, edgeData, nodeData } = this.state;
+
+    const data = { name: networkName };
     const elements = this.createCyElements(edgeData, nodeData);
 
-    if (elements && elements.length > 0) {
-      try {
-        this.controller.setNetwork(elements, { name: networkName });
-      } catch (e) {
-        console.log(e); // TODO Show error to user
-      }
+    // Create another Cytoscape instance, just so we can apply a layout
+    const cy = new Cytoscape();
+    cy.data(data);
+    cy.add(elements);
+
+    // Apply grid layout (but first, calculate an adequate bounding box where nodes don't overlap)
+    const nc = cy.nodes().length;
+    const size = 50 * Math.round(Math.sqrt(nc));
+    const w = 1.25 * size;
+    const h = 0.75 * size;
+    cy.layout({ name: 'grid', boundingBox: { x1: 0, y1: 0, w, h } }).run();
+    
+    // Get the json data that should now include the new node positions
+    const json = cy.json(true);
+    cy.destroy(); // We can now destroy this Cytoscape instance
+
+    // Ask the server to import the json data
+    const res = await fetch( `/api/document/json`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(json),
+    });
```

Comment:
In future, the layout could be running on the server in the `/api/document/json` route so that things are automatic and never forgotten (by the client).  For example, the server could apply a layout only if the payload doesn't contain positions.

Reply:
OK, I'll do that once I have the Cy3 Import working with the CX2 option, because when it imports a network that has no views, I think it will make more sense to apply the layout on server side anyway. Then I'll move this one to the server as well.